{
    "url": "https://api.github.com/repos/spack/spack/issues/2354",
    "repository_url": "https://api.github.com/repos/spack/spack",
    "labels_url": "https://api.github.com/repos/spack/spack/issues/2354/labels{/name}",
    "comments_url": "https://api.github.com/repos/spack/spack/issues/2354/comments",
    "events_url": "https://api.github.com/repos/spack/spack/issues/2354/events",
    "html_url": "https://github.com/spack/spack/issues/2354",
    "id": 189846781,
    "node_id": "MDU6SXNzdWUxODk4NDY3ODE=",
    "number": 2354,
    "title": "Spack Crashes with Mystery Error",
    "user": {
        "login": "citibeth",
        "id": 1924215,
        "node_id": "MDQ6VXNlcjE5MjQyMTU=",
        "avatar_url": "https://avatars.githubusercontent.com/u/1924215?v=4",
        "gravatar_id": "",
        "url": "https://api.github.com/users/citibeth",
        "html_url": "https://github.com/citibeth",
        "followers_url": "https://api.github.com/users/citibeth/followers",
        "following_url": "https://api.github.com/users/citibeth/following{/other_user}",
        "gists_url": "https://api.github.com/users/citibeth/gists{/gist_id}",
        "starred_url": "https://api.github.com/users/citibeth/starred{/owner}{/repo}",
        "subscriptions_url": "https://api.github.com/users/citibeth/subscriptions",
        "organizations_url": "https://api.github.com/users/citibeth/orgs",
        "repos_url": "https://api.github.com/users/citibeth/repos",
        "events_url": "https://api.github.com/users/citibeth/events{/privacy}",
        "received_events_url": "https://api.github.com/users/citibeth/received_events",
        "type": "User",
        "site_admin": false
    },
    "labels": [
        {
            "id": 480046049,
            "node_id": "MDU6TGFiZWw0ODAwNDYwNDk=",
            "url": "https://api.github.com/repos/spack/spack/labels/UI",
            "name": "UI",
            "color": "d4c5f9",
            "default": false,
            "description": null
        },
        {
            "id": 446623247,
            "node_id": "MDU6TGFiZWw0NDY2MjMyNDc=",
            "url": "https://api.github.com/repos/spack/spack/labels/unreproducible",
            "name": "unreproducible",
            "color": "fbca04",
            "default": false,
            "description": null
        }
    ],
    "state": "closed",
    "locked": false,
    "assignee": null,
    "assignees": [],
    "milestone": null,
    "comments": 1,
    "created_at": "2016-11-16T21:13:31Z",
    "updated_at": "2017-11-22T09:45:31Z",
    "closed_at": "2017-11-22T09:45:31Z",
    "author_association": "MEMBER",
    "active_lock_reason": null,
    "body": "The following was caused by a bad/garbage `package.py` file.  This should really have a better error message.\r\n\r\nAnyone know what is going wrong?  `spack reindex` did not fix.\r\n\r\n```\r\n$ spack install modele-control\r\nTraceback (most recent call last):\r\n  File \"/home2/rpfische/spack4/bin/spack\", line 203, in <module>\r\n    main()\r\n  File \"/home2/rpfische/spack4/bin/spack\", line 180, in main\r\n    return_val = command(parser, args)\r\n  File \"/home2/rpfische/spack4/lib/spack/spack/cmd/install.py\", line 313, in install\r\n    specs = spack.cmd.parse_specs(args.package, concretize=True)\r\n  File \"/home2/rpfische/spack4/lib/spack/spack/cmd/__init__.py\", line 108, in parse_specs\r\n    spec.concretize()  # implies normalize\r\n  File \"/home2/rpfische/spack4/lib/spack/spack/spec.py\", line 1330, in concretize\r\n    self._expand_virtual_packages(),\r\n  File \"/home2/rpfische/spack4/lib/spack/spack/spec.py\", line 1247, in _expand_virtual_packages\r\n    spec)\r\n  File \"/home2/rpfische/spack4/lib/spack/spack/concretize.py\", line 122, in choose_virtual_or_external\r\n    candidates = self._valid_virtuals_and_externals(spec)\r\n  File \"/home2/rpfische/spack4/lib/spack/spack/concretize.py\", line 62, in _valid_virtuals_and_externals\r\n    providers = spack.repo.providers_for(spec)\r\n  File \"/home2/rpfische/spack4/lib/spack/spack/repository.py\", line 75, in converter\r\n    return function(self, spec_like, *args, **kwargs)\r\n  File \"/home2/rpfische/spack4/lib/spack/spack/repository.py\", line 240, in providers_for\r\n    providers = self.provider_index.providers_for(vpkg_spec)\r\n  File \"/home2/rpfische/spack4/lib/spack/spack/repository.py\", line 234, in provider_index\r\n    self._provider_index.merge(repo.provider_index)\r\n  File \"/home2/rpfische/spack4/lib/spack/spack/repository.py\", line 645, in provider_index\r\n    self._update_provider_index()\r\n  File \"/home2/rpfische/spack4/lib/spack/spack/repository.py\", line 639, in _update_provider_index\r\n    self._provider_index.to_yaml(new)\r\n  File \"/home2/rpfische/spack4/lib/spack/llnl/util/lock.py\", line 312, in __exit__\r\n    if self._as.__exit__(type, value, traceback):\r\n  File \"/home2/rpfische/spack4/lib/spack/spack/file_cache.py\", line 152, in __exit__\r\n    raise value\r\nAttributeError\r\n```\r\n",
    "performed_via_github_app": null
}